### PR TITLE
[REM3-326] Restore Remoting and Endpoint.connect(URI, OptionMap, Call…

### PR DESCRIPTION
**Pull request by bmaxwell**
…backHandler)

https://issues.jboss.org/browse/REM3-326

### CODE REVIEW

File: src/main/java/org/jboss/remoting3/EndpointImpl.java [hunk 1]
```diff
@@ -628,6 +629,27 @@ public boolean setResult(final ConnectionHandlerFactory connHandlerFactory) {
...
+	 * See JBEAP-14783
+	 */
+	@Deprecated
+	public IoFuture<Connection> connect(final URI destination, final OptionMap connectOptions,
+			final CallbackHandler callbackHandler) throws IOException {
+		final AuthenticationContextConfigurationClient client = AUTH_CONFIGURATION_CLIENT;
+		AuthenticationContext authenticationContext = AuthenticationContext.captureCurrent();
+		AuthenticationConfiguration connectionConfiguration = client
+				.getAuthenticationConfiguration(destination, authenticationContext).useCallbackHandler(callbackHandler);
+		RemotingOptions.mergeOptionsIntoAuthenticationConfiguration(connectOptions, connectionConfiguration);
```
Comment (dmlloyd) [inline src/main/java/org/jboss/remoting3/EndpointImpl.java]: This needs to be like:
```java
connectionConfiguration = RemotingOptions.mergeOptionsIntoAuthenticationConfiguration(connectOptions, connectionConfiguration);
```

File: src/main/java/org/jboss/remoting3/Endpoint.java [hunk 2]
```diff
@@ -379,6 +380,8 @@ static Endpoint getCurrent() {
      */
     IoFuture<Connection> connect(URI destination, InetSocketAddress bindAddress, OptionMap connectOptions, SSLContext sslContext, AuthenticationConfiguration connectionConfiguration);
 
+    IoFuture<Connection> connect(final URI destination, final OptionMap connectOptions, final CallbackHandler callbackHandler) throws IOException;
```
Comment (dmlloyd) [inline src/main/java/org/jboss/remoting3/Endpoint.java]: These don't need to be final.  IDEs add it sometimes; it's not clear why though :)

Comment (bmaxwell) [reply src/main/java/org/jboss/remoting3/Endpoint.java]: Updated

Comment (dmlloyd) [reply src/main/java/org/jboss/remoting3/Endpoint.java]: Does not seem to be?

**Pull request merged**